test/cluster/mzcompose.py

```diff
@@ -2381,3 +2383,43 @@ def validate_since(since: int, name: str) -> None:
     table_since2, index_since2 = collect_sinces()
     validate_since(table_since2, "table_since2")
     validate_since(index_since2, "index_since2")
+
+
+def workflow_test_concurrent_connections(c: Composition) -> None:
```

Comment:
@ParkMyCar I put my explicit test in here directly here, hope that's fine for you. CC @philip-stoev 

This is based on Nikhil's suggestion in https://github.com/MaterializeInc/materialize/issues/21782#issuecomment-1722067303, but we don't hit the 0.15 and 0.30 s milestones yet, so the test could still be changed later if things get better. On main this fails brutally.

Reply:
Put it in Nightyl when ready.

Reply:
This will be auto-enabled in regular tests pipeline when this PR lands. It only takes a few seconds, so no need for nightly I think.

Reply:
Oh sorry, you are right line 89 enables it.

Reply:
Thank you for adding this @def-! I'm quite surprised, but very happy, to see this test passes!